src/wallet/rpcwallet.cpp

```diff
@@ -4132,6 +4211,181 @@ UniValue walletcreatefundedpsbt(const JSONRPCRequest& request)
+                    "                                      the number of addresses.\n",
+                    "  \"hex\" : \"value\",       (string) If add_to_wallet is false, the hex-encoded raw transaction with signature(s)\n"
+                    "  \"psbt\" : \"value\",      (string) If more signatures are needed, or if add_to_wallet is false, the base64-encoded (partially) signed transaction\n"
+                    "}\n"
+                        },
+                        RPCExamples{
+                    "\nCreate a transaction with a specific input\n"
+                    + HelpExampleCli("send", "'{\"1PGFqEzfmQch1gKD3ra4k18PNj3tTUUSqg\": 0.1}' '[{\"txid\":\"a08e6907dbbd3d809776dbfc5d82e371b764ed838b5655e72f463568df1aadf0\", \"vout\":1}]'") +
+                    "\nSpecify the order of outputs, set fee rate to 1 satoshi per byte and return result without adding to wallet or broadcasting to the network\n"
+                    + HelpExampleCli("send", "'[{\"1PGFqEzfmQch1gKD3ra4k18PNj3tTUUSqg\": 0.1}, {\"1LtvqCaApEdUGFkpKMM4MstjcaL4dKg8SP\": 0.1}]' [] '{\"feeRate\": 0.00001, \"add_to_wallet\": false}'")
+                        },
+                    }.Check(request);
+
+    RPCTypeCheck(request.params, {
+        UniValueType(), // ARR or OBJ, checked later
+        UniValue::VARR,
+        UniValue::VOBJ,
+        }, true
+    );
+
+    CAmount fee;
+    int change_position;
+    bool rbf = pwallet->m_signal_rbf;
+    const UniValue &replaceable_arg = request.params[2]["replaceable"];
+    if (!replaceable_arg.isNull()) {
+        RPCTypeCheckArgument(replaceable_arg, UniValue::VBOOL);
+        rbf = replaceable_arg.isTrue();
+    }
+    CMutableTransaction rawTx = ConstructTransaction(request.params[1], request.params[0], request.params[2]["locktime"], rbf);
+    CCoinControl coin_control;
+    // Automatically select coins, unless at least one is manually selected. Can
+    // be overriden by options.add_inputs.
+    coin_control.m_add_inputs = rawTx.vin.size() == 0;
+    FundTransaction(pwallet, rawTx, fee, change_position, request.params[2], coin_control);
+
+    bool add_to_wallet = true;
+    if (request.params[2].exists("add_to_wallet"))
+        add_to_wallet = request.params[2]["add_to_wallet"].get_bool();
+
+    // Make a blank psbt
+    PartiallySignedTransaction psbtx(rawTx);
+
+    // Fill transaction with out data and attempt to sign
+    bool complete = true;
+    bool bip32derivs = pwallet->IsWalletFlagSet(WALLET_FLAG_EXTERNAL_SIGNER);
+
+    const TransactionError err = FillPSBT(pwallet, psbtx, complete, 1, true, bip32derivs);
+    if (err != TransactionError::OK) {
+        throw JSONRPCTransactionError(err);
+    }
+
+    if (pwallet->IsWalletFlagSet(WALLET_FLAG_EXTERNAL_SIGNER)) {
+        const std::string command = gArgs.GetArg("-signer", DEFAULT_EXTERNAL_SIGNER);
+        if (command == "") throw std::runtime_error(std::string(__func__) + ": restart bitcoind with -signer=<cmd>");
```

Comment:
What if there are multiple wallets, with different signers?

IMO `-signer` needs to be replaced with either a wallet-stored path, or a path provided when the wallet is loaded...

Reply:
Storing `signer` in the wallet makes sense, but I also think it can wait for a later PR.

Reply:
`3AhLfoVCK1UcA8VjCDu6r1ZQTLEdVq3LPc`

Reply:
`3AhLfoVCK1UcA8VjCDu6r1ZQTLEdVq3LPc`